src/ir/access_path_root.h

```diff
@@ -0,0 +1,137 @@
+#ifndef SRC_IR_ACCESS_PATH_ROOT_H_
+#define SRC_IR_ACCESS_PATH_ROOT_H_
+
+#include "absl/types/variant.h"
+#include "absl/strings/str_join.h"
+#include "src/common/logging/logging.h"
+#include "third_party/arcs/proto/manifest.pb.h"
+
+// The classes in this file describe the root of an AccessPath. At the moment,
+// we have only two types of roots: a HandleConnectionSpecAccessPathRoot and
+// a HandleConnectionAccessPathRoot, describing a HandleConnectionSpec in a
+// ParticleSpec and a Handle in a Particle, respectively.
+namespace raksha::ir {
+
+// A HandleConnectionSpecAccessPathRoot describes the root as a
+// HandleConnectionSpec in a ParticleSpec. We consider this uninstantiated as
+// ParticleSpecs are abstract Particles which may be instantiated in multiple
+// locations.
+class HandleConnectionSpecAccessPathRoot {
```

Comment:
OK for now, but we might want to think about non-Arcs centric name.

Reply:
Indeed. I want to generalize most of this stuff, but I want to study at least one other system before I do to get a better sense of what abstraction to use.